testing/Android_SDK_E2E_Test suite.md

```diff
@@ -0,0 +1,71 @@
+
+### UiAutomator
+
+UiAutomator is another java based framework that is favoured and documented on the Android developers website [2].  This framework is used to build test suites that can perform actions on both user and system apps.  Like the Espresso framework, UiAutomator does not allow for switching between the user application and the browser, again favouring the use of intents to assert that the call was made. 
+
+### Appium
+
+Appium is a cross platform testing tool that allows for testing of native and hybrid applications on multiple platforms such as iOS and Android.  Appium has a number of clients that can be used to write tests in languages such as NodeJS, Java, Python, Ruby, etc. 
+
+### WD.js
+
+This client has many positives - there is a large number of locator strategies to target elements in views, the syntax of such tests can be written in a promise chain, and its possible to use mochas transfer promisness to make assertions a natural part of promise chain.  However, despite the positive features, the Wd.js client cannot handle a redirect to the browser at the present time - the problem seems to be that all chromedriver sessions are terminated when attempting to switch context from native to webview which cause the tests to crash. 
+
+### Webdriverio (nodejs)
+
+Like the WD.js library like the WD.js client, tests using this client can be written in a short chained syntax and can also be used in conjunction with features of tools like mochas transfer promisness to make assertions part of this chained syntax.  Unlike the WD.js client however, webdriverio can handle the redirect to the browser from the application, however there is only two possible locator strategies - xpath and accessibilityId. 
+
+### Java-Client (Appium Java SDK)
+
+Unlike the javascript libraries the java-client cannot be written in a promise chain syntax, however, there is a large number of locator strategies and it can handle the transition between native application and mobile browser.  Another major feature of the java client is the support for page object classes - using the following annotations, @FindBy(someStrategy), @AndroidFindBy(someStrategy) and @iOSFindBy(someStrategy), allows the creation of a single page object when creating tests for applications on multiple platforms that have the same content and flow but identifiers unique to their platform rather than a separate page object for each [3]. 
+
+## Framework Recommendation
+
+Appium is the framework of choice from those explored as it is the only framework explored that provides clients that can handle the redirect to the browser that is an integral test case we must cover.  Of the two clients, the Java-Client is likely to be the superior choice as it has a greater number of locator strategies and the annotation selector targeting provided can reduce the size of the codebase.
```

Comment:
It will be good to create ticket for IOS Swift support. 
More context:
https://discuss.appium.io/t/does-appium-support-swift-language/7980/3

Reply:
Actually we don't need to have it in Swift, it's client side language only, Appium Java SDK can test both Android and iOS.

Reply:
Any Appium Client can be used to test native, hybrid and mobile web apps on iOS, Android, and Windows mobile devices. A full list of clients available can be found here - https://appium.io/docs/en/about-appium/appium-clients/